openfisca_core/formulas.py

```diff
@@ -774,15 +766,11 @@ def new_filled_column(
     elif isinstance(source_file_path, str):
         source_file_path = source_file_path.decode('utf-8')
 
-    if start_date is UnboundLocalError:
-        start_date = None if reference_column is None else reference_column.start
-    elif start_date is not None:
-        assert isinstance(start_date, datetime.date)
-
-    if stop_date is UnboundLocalError:
-        stop_date = None if reference_column is None else reference_column.end
-    elif stop_date is not None:
-        assert isinstance(stop_date, datetime.date)
+    if end is UnboundLocalError:
+        end = None if reference_column is None else reference_column.end
+    elif end is not None:
+        assert isinstance(end, str)
+        get_datetime_date(name, end)  # Raises ValueError on format error.
```

Comment:
I don't see how you would need to call this helper several times if you just parsed the string once and for all.
Careful, here, we are parsing the `end` attribute _every single time_ we call `find_function`. 

It seems much more natural to parse the input only once. The fact that we end end using a method named `get_datetime_date` just to raise an error is a clear sign that our code structure is lopsided.